http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -179,30 +208,36 @@ public void onError(Throwable t) {
                 public void onComplete() {
                     try {
                         runAndLogError(String.format("Subscriber %s threw an exception in onComplete.", subscriber.toString()),
-                                       subscriber::onComplete);
+                                subscriber::onComplete);
                         requestContext.handler().complete();
                     } finally {
-                        finalizeRequest(requestContext, channelContext);
+                        // Run in the event loop to avoid sync issues in finalizeRequest
```

Comment:
I think we should be guaranteed to be running on the channel thread.

Reply:
Double checked this and looks like `HttpStreamsClientHandler` does guarantee this. Thanks for pointing it out!